libs/utils/android/workloads/videopreview.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+
+from subprocess import Popen, PIPE
+from android import Screen, Workload, System
+from time import sleep
+from devlib.utils.android import adb_command
+
+import logging
+
+GET_FRAME_STATS_CMD = 'shell dumpsys gfxinfo {} > {}'
+
+class VideoPreview(Workload):
+    """
+    Android VideoPreview workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.GoogleCamera'
+    activity = 'com.android.camera.CameraActivity'
+
+    # Setup logger
+    logger = logging.getLogger('VideoPreview')
+    logger.setLevel(logging.INFO)
+
+
+    def __init__(self, test_env):
+        super(VideoPreview, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'VideoPreview')
+
+    def run(self, exp_dir, preview_duration_s, collect=''):
+
+        # Initialize energy meter results
+        nrg_data, nrg_file = None, None
+
+        # Unlock device screen (assume no password required)
+        self.target.execute('input keyevent 82')
+        # Press Back button to be sure we run the video from the start
+        self.target.execute('input keyevent KEYCODE_BACK')
+
+        # Force screen in LANDSCAPE mode
+        Screen.set_orientation(self.target, portrait=False)
+
+        # Start VideoPreview video on the target device
+        self.target.execute('am start -a android.media.action.VIDEO_CAPTURE')
```

Comment:
Yes but that does `start -n`, this is `start -a` because an action is needed to directly start video capture mode. Otherwise we can use the start method of the API and then swipe to select recording mode.